prov/rxd/src/rxd_init.c

```diff
@@ -104,9 +104,8 @@ int rxd_info_to_rxd(uint32_t version, const struct fi_info *core_info,
 	*info->rx_attr = *rxd_info.rx_attr;
 	*info->ep_attr = *rxd_info.ep_attr;
 	*info->domain_attr = *rxd_info.domain_attr;
-	info->domain_attr->caps = ofi_pick_core_flags(rxd_info.domain_attr->caps,
-						core_info->domain_attr->caps,
-						FI_LOCAL_COMM | FI_REMOTE_COMM);
+	info->domain_attr->caps = rxd_info.domain_attr->caps &
+				  core_info->domain_attr->caps;
```

Comment:
I think the original code is correct here. We don't necessarily want to only pick the core caps. We want to pick only the core caps for the *_COMM caps which is what ofi_pick_core_flags is doing -  return all other util flags (not the "use_core_flags") ORed with the "use_core_flags" supported by the core provider.
Have you verified that this change fixes the issue?

Reply:
The original code overrides any setting of the util provider with what it received from the core provider. What is then the requirement for other flags which are not say supported by the core but not by the util providers. should the info be returned as supported?

Reply:
The original code should override the util settings only for the specified flags. For any flags not specified in the function (so in this case, anything other than the *COMM flags), it should select the util flags.
Right now there are only 3 defined domain caps: FI_LOCAL_COMM, FI_REMOTE_COMM, and FI_SHARED_AV. So what the function is saying is - use the rxd FI_SHARED_AV setting, whatever it is, and of the other 2 flags, select only the ones that are supported by the core provider. This is the correct behavior.
In your changes, if the core provider did not support FI_SHARED_AV but rxd did, the resulting caps would show it did not support it which would be incorrect.

Reply:
okay and so if util does not support a COMM setting but the core does, do we still need to return it as supported?

Reply:
Unlikely scenario since they both support both but if the util does not support it but the core does, it probably should not return that flag.